### PR TITLE
Remove 2122 representative user support

### PR DESCRIPTION
**Pull request by gabezurita**
As per the production user type rollout process, the Identity Team (@department-of-veterans-affairs/octo-identity) has only approved support for 21a un-accredited representative users and has requested that ARF remove support for 2122 accredited representative users.

This commit removes the 2122 accredited representative user support until 2122 users have completed the necessary review process.

This commit also adds @department-of-veterans-affairs/octo-identity as owners of modules/accredited_representative_portal/spec/services/accredited_representative_portal/representative_user_loader_spec.rb

The above means that the Identity Team will be notified of any changes to the file and can review them before they are merged.

Additionally, the more manual ARP Pilot representative user approach has been rejected in favor of pursuing a more automated user validation approach. As a result, I am removing any PilotRepresentative code. The automated rep user validation work will be done in the below epic:
https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/75746

## Summary

- *This work is behind a feature toggle (flipper): YES*

## Related Issues
* https://app.zenhub.com/workspaces/accredited-representative-facing-team-65453a97a9cc36069a2ad1d6/issues/gh/department-of-veterans-affairs/va.gov-team/90755
* DSCA OCTO Slack document: https://dsva.slack.com/docs/T03FECE8V/F07GUSS5CT

## Testing done

- [X] *New code is covered by unit tests*

## What areas of the site does it impact?
* ARP Portal Login

## Note to the reviewer
I'll resolve schema conflicts once I've gotten a review!

### CODE REVIEW

File: .github/CODEOWNERS
Comment (nihil2501) [reply .github/CODEOWNERS:2126]: Potentially also add identity as a codeowner for `pilot_representative.rb` (see my PR review-wide comment).

**Review by nihil2501 — CHANGES_REQUESTED** [review]
Add identity as codeowner to the user loader code (and possibly also the pilot representative code (see below comment)) as well?

I'm wondering though whether it might be more correct on the whole to not delete this code but instead defeat it at its only integration point. Which is to say that we can modify the behavior of the pilot representative lookup with a feature flag or a different implementation change and also include a comment.

The benefit of this would be if that code is correct then we can save a small / medium-ish amount of correct code with no real downside.

If we do on the other hand want to delete it, then this change LGTM after codeowner updates.

**Comment by nihil2501** [comment]
@gabezurita Ah I see in the PR description we wanted to change business logic anyway, so it's just the codeowners and safe migration points to possibly address for me.

**Review by bramleyjl — CHANGES_REQUESTED** [review]
Adding the Identity team as codeowners of the `representative_user_loader` is a good idea, that class is based off of the `user_loader` that we maintain.

As for the pilot representative code, if it isn't being used then it should not be in the `vets-api` main branch, we'd like to keep the codebase as clean as possible by not hanging on to unused code.

It's also cleaner and safer to submit removal PRs as first the business logic removal and then the DB migration; the Identity team would prefer it if you re-submitted this without the migrations.

**Comment by bosawt** [comment]
> Adding the Identity team as codeowners of the `representative_user_loader` is a good idea, that class is based off of the `user_loader` that we maintain.
> 
> As for the pilot representative code, if it isn't being used then it should not be in the `vets-api` main branch, we'd like to keep the codebase as clean as possible by not hanging on to unused code.
> 
> It's also cleaner and safer to submit removal PRs as first the business logic removal and then the DB migration; the Identity team would prefer it if you re-submitted this without the migrations.

In fact, the business logic/DB separated into different PRs is a Platform requirement, so it's not even up to the Identity team, you'll just have to do it anyways.

**Comment by gabezurita** [comment]
> Adding the Identity team as codeowners of the `representative_user_loader` is a good idea, that class is based off of the `user_loader` that we maintain.
> 
> As for the pilot representative code, if it isn't being used then it should not be in the `vets-api` main branch, we'd like to keep the codebase as clean as possible by not hanging on to unused code.
> 
> It's also cleaner and safer to submit removal PRs as first the business logic removal and then the DB migration; the Identity team would prefer it if you re-submitted this without the migrations.

I should have clarified, but I kept the migration PR in the initial PR so all the necessary changes could be viewed simultaneously. I'll split off the migration changes and tag you all.

File: .github/CODEOWNERS
Comment (gabezurita) [reply .github/CODEOWNERS:2126]: Ah, yep... I left out the `representative_user_loader.rb`; thanks for catching that. See commit: cfd20e6a35

File: lib/accredited_representation/seed_data.rb
Comment (ddzz) [inline lib/accredited_representation/seed_data.rb:1]: We could rename this file and method to something more descriptive, like `create_accredited_individual`.

Comment (gabezurita) [reply lib/accredited_representation/seed_data.rb:1]: Sure, here you go:
d3243e8cde

Comment (gabezurita) [reply lib/accredited_representation/seed_data.rb:1]: However, given the new 2122 approach, we could delete this file, too.

**Comment by gabezurita** [comment]
@nihil2501, @bramleyjl, @bosawt see below for migration PR:
https://github.com/department-of-veterans-affairs/vets-api/pull/18109

I've removed the migration files from this PR.

We're ready for a round of review.

**Review by bramleyjl — APPROVED** [review]

**Review by nihil2501 — APPROVED** [review]
LGTM

**Comment by gabezurita** [comment]
@department-of-veterans-affairs/backend-review-group and @department-of-veterans-affairs/va-api-engineers, this is ready for review. The migration PR is below:
https://github.com/department-of-veterans-affairs/vets-api/pull/18109

**Review by ericboehs — APPROVED** [review]

**Pull request merged**
